phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -2957,6 +3010,11 @@ MutationState dropTable(String schemaName, String tableName, String parentTableN
                 Delete linkDelete = new Delete(linkKey, clientTimeStamp);
                 tableMetaData.add(linkDelete);
             }
+            if (tableType == PTableType.TABLE) {
+                // acquire a mutex on the table to prevent creating views while concurrently
+                // dropping the base table
+                acquiredMutex = writeCell(null, schemaName, tableName, null);
```

Comment:
Nothing is prevent even if `acquiredMutex` is false here. I think you should throw `ConcurrentTableMutationException` here as well? You might might wanna push down that logic to `writeCell()` method?